features/devices/isis/isis_route_install_test.go

```diff
@@ -0,0 +1,317 @@
+
+	rxIsisRrV4.Addresses().Add().
+		SetAddress(tc["rxAdvRouteV4"].(string)).
+		SetPrefix(32).
+		SetCount(tc["rxRouteCount"].(int32)).
+		SetStep(1)
+
+	rxIsisRrV6 := rxIsis.
+		V6Routes().
+		Add().SetName("rx_isis_rr6").SetLinkMetric(10)
+
+	rxIsisRrV6.Addresses().Add().
+		SetAddress(tc["rxAdvRouteV6"].(string)).
+		SetPrefix(32).
+		SetCount(tc["rxRouteCount"].(int32)).
+		SetStep(1)
+
+	for i := 1; i <= 4; i++ {
+		flow := c.Flows().Add()
+		flow.Duration().FixedPackets().SetPackets(tc["pktCount"].(int32))
+		flow.Rate().SetPps(tc["pktRate"].(int64))
+		flow.Size().SetFixed(tc["pktSize"].(int32))
+		flow.Metrics().SetEnable(true)
+	}
+
+	ftxV4 := c.Flows().Items()[0]
+	ftxV4.SetName("ftxV4")
+	ftxV4.TxRx().Device().
+		SetTxNames([]string{txIsisRrV4.Name()}).
+		SetRxNames([]string{rxIsisRrV4.Name()})
+
+	ftxV4Eth := ftxV4.Packet().Add().Ethernet()
+	ftxV4Eth.Src().SetValue(txEth.Mac())
+
+	ftxV4Ip := ftxV4.Packet().Add().Ipv4()
+	ftxV4Ip.Src().SetValue(tc["txAdvRouteV4"].(string))
+	ftxV4Ip.Dst().SetValue(tc["rxAdvRouteV4"].(string))
+
+	ftxV4Tcp := ftxV4.Packet().Add().Tcp()
+	ftxV4Tcp.SrcPort().SetValue(5000)
+	ftxV4Tcp.DstPort().SetValue(6000)
+
+	ftxV6 := c.Flows().Items()[1]
+	ftxV6.SetName("ftxV6")
+	ftxV6.TxRx().Device().
+		SetTxNames([]string{txIsisRrV6.Name()}).
+		SetRxNames([]string{rxIsisRrV6.Name()})
+
+	ftxV6Eth := ftxV6.Packet().Add().Ethernet()
+	ftxV6Eth.Src().SetValue(txEth.Mac())
+
+	ftxV6Ip := ftxV6.Packet().Add().Ipv6()
+	ftxV6Ip.Src().SetValue(tc["txAdvRouteV6"].(string))
+	ftxV6Ip.Dst().SetValue(tc["rxAdvRouteV6"].(string))
+
+	ftxV6Tcp := ftxV6.Packet().Add().Tcp()
+	ftxV6Tcp.SrcPort().SetValue(5000)
+	ftxV6Tcp.DstPort().SetValue(6000)
+
+	frxV4 := c.Flows().Items()[2]
+	frxV4.SetName("frxV4")
+	frxV4.TxRx().Device().
+		SetTxNames([]string{rxIsisRrV4.Name()}).
+		SetRxNames([]string{txIsisRrV4.Name()})
+
+	frxV4Eth := frxV4.Packet().Add().Ethernet()
+	frxV4Eth.Src().SetValue(rxEth.Mac())
+
+	frxV4Ip := frxV4.Packet().Add().Ipv4()
+	frxV4Ip.Src().SetValue(tc["rxAdvRouteV4"].(string))
+	frxV4Ip.Dst().SetValue(tc["txAdvRouteV4"].(string))
+
+	frxV4Tcp := frxV4.Packet().Add().Tcp()
+	frxV4Tcp.SrcPort().SetValue(6000)
+	frxV4Tcp.DstPort().SetValue(5000)
+
+	frxV6 := c.Flows().Items()[3]
+	frxV6.SetName("frxV6")
+	frxV6.TxRx().Device().
+		SetTxNames([]string{rxIsisRrV6.Name()}).
+		SetRxNames([]string{txIsisRrV6.Name()})
+
+	frxV6Eth := frxV6.Packet().Add().Ethernet()
+	frxV6Eth.Src().SetValue(rxEth.Mac())
+
+	frxV6Ip := frxV6.Packet().Add().Ipv6()
+	frxV6Ip.Src().SetValue(tc["rxAdvRouteV6"].(string))
+	frxV6Ip.Dst().SetValue(tc["txAdvRouteV6"].(string))
+
+	frxV6Tcp := frxV6.Packet().Add().Tcp()
+	frxV6Tcp.SrcPort().SetValue(6000)
+	frxV6Tcp.DstPort().SetValue(5000)
+
+	api.Testing().Logf("Config:\n%v\n", c)
+	return c
+}
+
+func isisRouteInstallMetricsOk(api *otg.OtgApi, tc map[string]interface{}) bool {
+	for _, m := range api.GetIsIsMetrics() {
+		if m.L1SessionsUp() != 1 || m.L2SessionsUp() != 1 ||
```

Comment:
can we not hardcode these values ?

why are we checking L1/L2 database size btw ? Please note, this check does not ensure that routes are correctly advertised. Hence, for that, one MUST rely on isis states (not supported as of now).

so would suggest adding comments accordingly.

Reply:
Added TODO Comment